<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>simple_tcp_serverClass</class>
 <widget class="QMainWindow" name="simple_tcp_serverClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>884</width>
    <height>623</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>simple_tcp_server</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>421</width>
      <height>161</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="2">
      <widget class="QPushButton" name="btn_start">
       <property name="text">
        <string>start</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QPushButton" name="btn_send">
       <property name="text">
        <string>send</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="le_port">
       <property name="text">
        <string>12800</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="le_msg">
       <property name="text">
        <string>Hello World!</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>*Port</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QPushButton" name="btn_stop">
       <property name="text">
        <string>stop</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>*TCP/IP</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="le_addr">
       <property name="text">
        <string>192.168.31.8</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Message</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Clients</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="le_client_nums">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QPushButton" name="btn_get_clients">
       <property name="text">
        <string>clients nums</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPlainTextEdit" name="pte_commands">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>421</width>
      <height>371</height>
     </rect>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="pte_recv">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>220</y>
      <width>421</width>
      <height>371</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Commands:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>180</y>
      <width>141</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Recived from Clients:</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="pte_instruction">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>10</y>
      <width>421</width>
      <height>161</height>
     </rect>
    </property>
    <property name="plainText">
     <string>说明：
1. 点击&lt;start&gt; 使用TCP/IP和Port口启动一个服务，并开始监听
2. 点击&lt;stop&gt; 停止服务器，并且清空所有的客户端，置于可以重新开始服务的状态
3. 点击&lt;send&gt;将Message中的内容发送给列表中所有的客户端
4. 点击&lt;active clients&gt;返回连入的客户端的数量
Command窗口显示服务器的指令信息
Recived from Clients显示客户端发送给服务器的信息
支持多客户端的连入</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>884</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="simple_tcp_server.qrc"/>
 </resources>
 <connections/>
</ui>
